RELEASE/scripts/autoscend/auto_pre_adv.ash

```diff
@@ -325,11 +325,113 @@ void handlePreAdventure(location place)
 		}
 	}
 
+
+	// ML adjustment zone section
+	boolean doML = true;
+	boolean removeML = false;
+		// removeML MUST be true for purgeML to be used. This is only used for -ML locations like Smut Orc, and you must have 5+ SGEAs to use.
+		boolean purgeML = false;
+
+	boolean[location] highMLZones = $locations[Oil Peak, The Typical Tavern Cellar, The Haunted Boiler Room];
+	boolean[location] lowMLZones = $locations[The Smut Orc Logging Camp];
+
+	// Generic Conditions
+	if(get_property("kingLiberated").to_boolean())
+	{
+		doML = false;
+	}
+
+	if(((get_property("flyeredML").to_int() > 9999) || get_property("auto_hippyInstead").to_boolean() || (get_property("auto_war") == "finished") || (get_property("sidequestArenaCompleted") != "none")) && ((my_level() >= 13)))
+	{
+		doML = false;
+	}
+
+	// Item specific Conditions
+	if((equipped_amount($item[Space Trip Safety Headphones]) > 0) || (equipped_amount($item[Red Badge]) > 0))
+	{
+		doML = false;
+		removeML = true;
+	}
+
+	// Location Specific Conditions
+	if(lowMLZones contains place)
+	{
+		doML = false;
+		removeML = true;
+		purgeML = true;
+	}
+	if(highMLZones contains place)
+	{
+		doML = true;
+		removeML = false;
+	}
+
+	// Act on ML settings
+	if(doML)
+	{
+		// Catch when we leave lowMLZone, allow for being "side tracked" buy delay burning
+		if((have_effect($effect[Driving Intimidatingly]) > 0) && (get_property("auto_debuffAsdonDelay") >= 2))
+		{
+			print("No Reason to delay Asdon Usage");
+			uneffect($effect[Driving Intimidatingly]);
+			set_property("auto_debuffAsdonDelay", 0);
+		}
+		else if((have_effect($effect[Driving Intimidatingly]).to_int() == 0)  && (get_property("auto_debuffAsdonDelay") >= 0))
+		{
+			set_property("auto_debuffAsdonDelay", 0);
+		}
+		else
+		{
+			set_property("auto_debuffAsdonDelay", get_property("auto_debuffAsdonDelay").to_int() + 1);
+			print("Delaying debuffing Asdon: " + get_property("auto_debuffAsdonDelay"));
+		}
+
+		auto_MaxMLToCap(150, false);
+	}
+
+	// If we are in some state where we do not want +ML (Level 13 or Smut Orc) make sure ML is removed
```

Comment:
If i am parsing this correctly, my_level() only sets doML = false so ML isnt added, but doesnt remove ML. Which is probably fine, just making sure your comment doesnt mean you wanted to remove ML but missed it.

Reply:
Originally yes, though after making that other change you pointed out above (>= to ==) I have now made it remove (not purge) in order to give us a better chance of hitting that Instant Karma.